### ISSUE TITLE
Browser: HTTP request header injection through links with linebreaks

### ISSUE THREAD

**Issue by xTibor** [SerenityOS/serenity#3633]
**Reproduction:**

```html
<!DOCTYPE html>
<html>
<head>
</head>
<body>
    <a href="http://192.168.1.103:9999/test HTTP/1.1
Abcd: efgh
Ijkl: mnop
">Click me</a>
</body>
</html>
```

**Request sent:**
```
GET /test HTTP/1.1
Abcd: efgh
Ijkl: mnop
 HTTP/1.1
Host: 192.168.1.103
User-Agent: Mozilla/4.0 (SerenityOS; x86) LibWeb+LibJS (Not KHTML, nor Gecko) LibWeb
Connection: close
```

**Comment by Lubrsi** [comment]
As reference, other browsers just encode the entire thing as a URL:
```
http://192.168.1.103:9999/test%20HTTP/1.1Abcd:%20efghIjkl:%20mnop
```

**Comment by bcoles** [comment]
This has been fixed. Presumably fixed in #6557.

```
# nc -lvp 1337 
Ncat: Version 7.91 ( https://nmap.org/ncat )
Ncat: Listening on :::1337
Ncat: Listening on 0.0.0.0:1337
Ncat: Connection from 10.1.1.101.
Ncat: Connection from 10.1.1.101:46579.
GET /test%20HTTP/1.1%0AAbcd:%20efgh%0AIjkl:%20mnop%0A HTTP/1.1
Host: 10.1.1.115
User-Agent: Mozilla/4.0 (SerenityOS; x86) LibWeb+LibJS (Not KHTML, nor Gecko) LibWeb
Accept-Encoding: gzip, deflate
Connection: close

```

```
WebContent(29): Web::EventHandler: Clicking on a link to http://10.1.1.115:1337/test HTTP/1.1
Abcd: efgh
Ijkl: mnop

WebContent(29): FrameLoader::load: http://10.1.1.115:1337/test HTTP/1.1
Abcd: efgh
Ijkl: mnop

```
